src/get-started/flutter-for/ios-devs.md

```diff
@@ -490,200 +859,593 @@ class SignaturePainter extends CustomPainter {
-        '/c': (context) => const MyPage(title: 'page C'),
-      },
-    ),
-  );
-}
-```
+```swift
+@Environment(\.openURL) private var openUrl
 
-Navigate to a route by `push`ing its name to the `Navigator`.
+// View code goes here
 
-<?code-excerpt "lib/intent.dart (Push)"?>
-```dart
-Navigator.of(context).pushNamed('/b');
+ Button("Open website") {
+      openUrl(
+        URL(
+          string: "https://google.com"
+        )!
+      )
+    }
 ```
 
-The `Navigator` class handles routing in Flutter and is used to get
-a result back from a route that you have pushed on the stack.
-This is done by `await`ing on the `Future` returned by `push()`.
+To achieve the same results in Flutter, you use the [`url_launcher`][] plugin. 
 
-For example, to start a ‘location’ route that lets the user select their
-location, you might do the following:
+<nav class="navbar bg-primary">
+ <ul class="navbar-nav navbar-code ml-auto">
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.repo.this}}/{{sample_path}}/lib/openapp.dart">View on Github</a>
+  </li>
+</ul>
+</nav>
 
-<?code-excerpt "lib/intent.dart (PushAwait)"?>
+<?code-excerpt "lib/openapp.dart (OpenAppExample)"?>
 ```dart
-Object? coordinates = await Navigator.of(context).pushNamed('/location');
+CupertinoButton(
+  onPressed: () async {
+    await launchUrl(
+      Uri.parse('https://google.com'),
+    );
+  },
+  child: const Text(
+    'Open website',
+  ),
+),
 ```
 
-And then, inside your ‘location’ route, once the user has selected their
-location, `pop()` the stack with the result:
-
-<?code-excerpt "lib/intent.dart (Pop)"?>
+## Themes, styles and media
+Flutter applications are easy to style; you can easily switch 
+between light and dark themes, 
+change the style of your text and UI components, 
+and more. This section covers aspects of styling your Flutter apps 
+and compares how you might do the same in SwiftUI.
+
+### Using dark mode
+In SwiftUI, if you want to use dark mode for your views, 
+you can call the `preferredColorScheme()` function that is implemented 
+on a `View`.
+    
+In Flutter, you control light and dark mode at the app-level 
+using the `theme` property of either your `App` class as follows:
+
+
+<nav class="navbar bg-primary">
+ <ul class="navbar-nav navbar-code ml-auto">
+ <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.dartpad}}/?id=c446775c3224787e51fb18b054a08a1c">Test in DartPad</a>
+  </li>
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.repo.this}}/{{sample_path}}/lib/cupertino_themes.dart">View on Github</a>
+  </li>
+</ul>
+</nav>
+
+<?code-excerpt "lib/cupertino_themes.dart (Theme)"?>
 ```dart
-Navigator.of(context).pop({'lat': 43.821757, 'long': -79.226392});
+    CupertinoApp(
+  theme: CupertinoThemeData(
+    brightness: Brightness.dark,
+  ),
+  home: HomePage(),
+);
 ```
 
-### How do I navigate to another app?
+### Styling text
```

Comment:
Material 3 introduced a LOT of new styling options, including a theming tool. Does any of this doc address that?

I'm concerned that this might already be out of date.

Reply:
@esouthren do happen to know if the CupertinoThemeData API will still use brightness?

Reply:
Thanks!